google-cloud-logging/lib/google/cloud/logging/service.rb

```diff
@@ -279,6 +285,24 @@ def default_options
           Google::Gax::CallOptions.new kwargs: default_headers
         end
 
+        ##
+        # @private Get a Google::Protobuf::Timestamp object from a Time object.
+        def time_to_timestamp time
+          return nil if time.nil?
+          # Make sure we have a Time object
+          return nil unless time.respond_to? :to_time
```

Comment:
I wonder if raising an error is a better choice here. Since start_time and end_time are optional parameters, passing an unsupported type could lead to the user to believe the argument was successfully applied. It is not hard to imagine someone trying to pass a time value as Fixnum, for example.